File: node.js/src/original/domain/books/entities/translator.ts
import { DeepReadonly, NonEmptyString, NonEmptyUuid } from '#core/typing';

export type TranslatorId = NonEmptyUuid<'TranslatorId'>;
export type TranslatorName = NonEmptyString<'TranslatorName'>;

export type Translator = DeepReadonly<{
  Id: TranslatorId;
  Name: TranslatorName;
}>;

File: csharp/Original/PublishingHouse.Domain/Books/Entities/Translator.cs
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Books.Entities;

public record Translator(TranslatorId Id, TranslatorName Name);
public record TranslatorId(Guid Value) : NonEmptyGuid(Value);
public record TranslatorName(string Value) : NonEmptyString(Value);
